test/unit/preprocessing/linear_dep_remover/test_sklearn_compat.py

```diff
@@ -7,16 +7,25 @@
 import numpy as np
 import pandas as pd
 import pytest
-from sklearn.utils.estimator_checks import parametrize_with_checks
 
 from fairlearn.preprocessing import CorrelationRemover
+from fairlearn.utils._fixes import parametrize_with_checks
+
+EXPECTED_FAILED_CHECKS = {
+    "CorrelationRemover": {
+        "check_transformer_data_not_an_array": (
+            "this estimator only accepts pandas dataframes or numpy ndarray as input."
+        ),
+    },
+}
```

Comment:
sklearn compatibility tests and these dicts can probablay be moved to a single file, but a different PR.

Reply:
I have a new PR today when I remove this one, and the next one I should remove the remaining so we might not need to :) but the adapted code should stay there anyway for the future. thanks for reviewing and merging today 🥳